src/teleop/gui/src/components/Cameras.vue

```diff
@@ -0,0 +1,155 @@
+      <div class="camerainfo" v-for="i in camsEnabled.length" :key="i">
+        <CameraInfo v-if="camsEnabled[i-1]" v-bind:name="names[i-1]" v-bind:id="i-1" v-on:newQuality="changeQuality($event)" v-on:swapStream="swapStream($event)" v-bind:stream="getStreamNum(i-1)"></CameraInfo>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  import Vue from 'vue'
+  import ROSLIB from 'roslib/src/RosLib'
+  import CameraSelection from '../components/CameraSelection.vue'
+  import CameraInfo from '../components/CameraInfo.vue'
+  
+  export default {
+    data() {
+      return {
+        camsEnabled: new Array(9).fill(false),
+        names: Array.from({length: 9}, (_,i) => "Camera: " + i),
+        cameraIdx: 1,
+        cameraName: "",
+        capacity: 2,
+        qualities: new Array(9).fill(1),
+        streamOrder: [-1, -1, -1, -1]
+      }
+    },
+
+    props: {
+      numCams: {
+        type: Number,
+        required: true
+      },
+      primary: {
+        type: Boolean,
+        required: true
+      }
+
+    },
+
+
+    methods: {
+
+      setCamIndex: function (index) { //every time a button is pressed, it changes cam status and adds/removes from stream
+        Vue.set(this.camsEnabled, index, !this.camsEnabled[index]);
+        this.changeStream(index);
+        this.sendCameras();
+      },
+
+      sendCameras: function() { //sends cameras to a service to display on screen
+        var msgs = [];
+        for(var i = 0; i < 4; i++){
+          var camId = this.streamOrder[i];
+          var res = this.qualities[camId];
+          msgs.push(new ROSLIB.Message({device: camId, resolution: res})); //CameraCmd msg
+        }
+
+        var changeCamsService = new ROSLIB.Service({
+              ros : this.$ros,
+              name : 'change_cameras',
+              serviceType : 'ChangeCameras'
+          });
+
+        var request = new ROSLIB.ServiceRequest({primary: this.primary, camera_cmds: msgs});
+        changeCamsService.callService(request, (result) => {});
+
+      },
+
+      addCameraName: function() {
+        Vue.set(this.names, this.cameraIdx, this.cameraName);
+      },
+
+      changeQuality({index, value}){
+        Vue.set(this.qualities, index, value);
+      },
+
+      swapStream({prev, newest}){
+        var temp = this.streamOrder[prev];
+        Vue.set(this.streamOrder, prev, this.streamOrder[newest]);
+        Vue.set(this.streamOrder, newest, temp);
+        this.sendCameras();
+      },
+
+      changeStream(index){
+        const found = this.streamOrder.includes(index);
+        if(found){
+          this.streamOrder.splice(this.streamOrder.indexOf(index),1);
+          this.streamOrder.push(-1);
+        }
+        else Vue.set(this.streamOrder, this.streamOrder.indexOf(-1), index);
+      },
+
+      getStreamNum(index){
+        return this.streamOrder.indexOf(index);
+      }
+
+    },
+
+    computed: {
+      checkCapacity(){
```

Comment:
I don't think this is used? We have maxedOut in CameraSelection.vue which seems to do this